db/version_edit.h

```diff
@@ -241,6 +241,57 @@ struct LevelFilesBrief {
   }
 };
 
+using WalNumber = uint64_t;
+
+// Information of a new WAL tracked by VersionEdit.
+class AddedWal {
+ public:
+  AddedWal() = default;
+
+  AddedWal(WalNumber number, uint64_t bytes) : number_(number), bytes_(bytes) {}
+
+  WalNumber GetLogNumber() const { return number_; }
+
+  uint64_t GetSizeInBytes() const { return bytes_; }
+
+  void EncodeTo(std::string* dst) const;
+
+  Status DecodeFrom(Slice* src);
+
+  std::string DebugString() const;
+
+ private:
+  WalNumber number_;
+  uint64_t bytes_;
+};
+
+JSONWriter& operator<<(JSONWriter& jw, const AddedWal& wal);
```

Comment:
We should probably declare the other `operator<<` too here (also affects the other class).